File: Add-In/Scan/ReviveThis.Net.AddIn.Sample.CSharp/SampleAddOn.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.Composition;
using System.Threading.Tasks;
using ReviveThis.Enums;
using ReviveThis.Interfaces;
using ReviveThis.Sample.CSharp.Classes;

namespace ReviveThis.Sample.CSharp
{
  [Export(typeof (IDetectionAddIn))]
  public class SampleAddOn : IDetectionAddIn
  {

    public string Author
    {
      get { return @"C# Author"; }
    }

    private Version _version;

    public Version Version
    {
      get
      {
        if (_version != null)
          return _version;

        return _version = new Version(0, 0, 0, 1);
      }
    }

    public string Name
    {
      get { return @"Add-On Sample (C#)"; }
    }

    public string[] Description
    {
      get { return new[]
      {
        "A Sample Add-On for ReviveThis.Net written in C#",
        string.Empty,
         "Hey look, another line in the description."
      };}
    }

    public void Dispose()
    {
      throw new NotImplementedException();
    }

    public async Task<ICollection<IDetectionResultItem>> Scan()
    {
      //await Task.FromResult(0);

      //throw new NotImplementedException();

      var result = new List<IDetectionResultItem>
      {
        new SampleResult(ScanResultType.CustomAddIn, "This is from my Add-On Sample (C#)"),
        new SampleResult(ScanResultType.CustomAddIn, "This appears to be another result from my Add-On Sample (C#)")
      };

      return result;
    }

    public ScanResultType ResultType
    {
      get { return ScanResultType.CustomAddIn; }
    }
  }
}

File: Add-In/Scan/ReviveThis.Net.AddIn.Sample.VisualBasic/SampleAddOn.vb
'[Export](typeof (IScanModule))]
Imports System.ComponentModel.Composition
Imports ReviveThis.AddIn.Sample.VisualBasic.Classes
Imports ReviveThis.Enums
Imports System.Collections.ObjectModel
Imports ReviveThis.Interfaces
Imports System.Threading.Tasks


<Export(GetType(IDetectionAddIn))>
Public Class SampleAddOn
    'Note: You must have "<Export(GetType(IScanAddIn))>" in your Add-On, else ReviveThis.Net won't find/use it.

    Implements IDetectionAddIn

    Public ReadOnly Property Author() As String Implements IModuleInformation.Author
        Get
            Return "VB.Net Author"
        End Get
    End Property

    Public ReadOnly Property Version() As Version Implements IModuleInformation.Version
        Get
            Return New Version(0, 0, 0, 1)
        End Get
    End Property

    Public ReadOnly Property Name() As String Implements IModuleInformation.Name
        Get
            Return "Add-On Sample (Visual Basic)"
        End Get
    End Property

    Public ReadOnly Property Description() As String() Implements IModuleInformation.Description
        Get
            Return New String() {"A Sample Add-On for ReviveThis.Net written in Visual Basic .Net", vbNullString, "Hey look, another line in the description."}
        End Get
    End Property

    Public Sub Dispose() Implements IDisposable.Dispose
        'We don't have anything to Dispose
    End Sub

    Public Async Function Scan() As Task(Of ICollection(Of IDetectionResultItem)) Implements IDetectionAddIn.Scan
        Dim result As New Collection(Of IDetectionResultItem)
        'Await Task.FromResult(0)
        result.Add(New SampleResult(ScanResultType.CustomAddIn, "This is from my Add-On Sample (Visual Basic)"))
        result.Add(New SampleResult(ScanResultType.CustomAddIn, "This appears to be another result from my Add-On Sample (Visual Basic)"))
        Return result
    End Function

    Public ReadOnly Property SortIndex() As ScanResultType Implements IDetectionAddIn.ResultType
        Get
            'The expected ScanResultType
            Return ScanResultType.CustomAddIn
        End Get
    End Property

End Class
